src/bootstrap/src/lib.rs

```diff
@@ -661,28 +661,32 @@ impl Build {
         &self.config.rust_info
     }
 
-    /// Gets the space-separated set of activated features for the standard
-    /// library.
+    /// Gets the space-separated set of activated features for the standard ibrary. This can be configured with the `std-features` key in config.toml.
     fn std_features(&self, target: TargetSelection) -> String {
-        let mut features = " panic-unwind".to_string();
+        let mut features: BTreeSet<&str> =
+            self.config.rust_std_features.iter().map(|s| s.as_str()).collect();
+
+        features.insert("panic-unwind");
 
         match self.config.llvm_libunwind(target) {
-            LlvmLibunwind::InTree => features.push_str(" llvm-libunwind"),
-            LlvmLibunwind::System => features.push_str(" system-llvm-libunwind"),
-            LlvmLibunwind::No => {}
-        }
+            LlvmLibunwind::InTree => features.insert("llvm-libunwind"),
+            LlvmLibunwind::System => features.insert("system-llvm-libunwind"),
+            LlvmLibunwind::No => false,
+        };
+
         if self.config.backtrace {
-            features.push_str(" backtrace");
+            features.insert("backtrace");
         }
         if self.config.profiler_enabled(target) {
-            features.push_str(" profiler");
+            features.insert("profiler");
         }
         // Generate memcpy, etc.  FIXME: Remove this once compiler-builtins
         // automatically detects this target.
         if target.contains("zkvm") {
-            features.push_str(" compiler-builtins-mem");
+            features.insert("compiler-builtins-mem");
         }
```

Comment:
We need to clarify something here. Should we run this logic only if the user hasn't set `std-features`, or should we run it regardless? I'm not sure which approach the library team would prefer.

cc @rust-lang/libs

Reply:
@onur-ozkan any thoughts on what should we do here? should I reach out to a channel on Zulip for feedback on this?

Reply:
We still aren't sure what the stability story behind standard library features is, so for now all of these are to be considered unstable and subject to change at any time. I'll leave it as your decision as an actual user of `rust.std-features` whether you think the current auto-detection logic is always useful or whether you need the flexibility of disabling certain auto-enabled library features.